src/main/java/sg/ncl/MainController.java

```diff
@@ -5273,4 +5313,230 @@ private TeamQuota extractTeamQuotaInfo(String responseBody) {
+        model.addAttribute("addMemberForm", new AddMemberForm());
+        return "add_member";
+    }
+
+    /**
+     * For a class project leader to add students to his class using their emails
+     *
+     * @param teamId
+     * @param addMemberForm
+     * @param redirectAttributes
+     * @return
+     * @throws WebServiceRuntimeException
+     */
+    @RequestMapping(value="/add_member/{teamId}", method= RequestMethod.POST)
+    public String addMember(@PathVariable String teamId, @Valid AddMemberForm addMemberForm,
+                            final RedirectAttributes redirectAttributes)  throws WebServiceRuntimeException {
+
+        log.info("Adding members to team {}", teamId);
+
+        String emails[] = addMemberForm.getEmails().split("\\r?\\n");
+
+        for (int i = 0; i < emails.length; i++){
+            if (!VALID_EMAIL_ADDRESS_REGEX.matcher(emails[i]).matches()) {
+                redirectAttributes.addFlashAttribute(MESSAGE, EMAIL_ADDRESS_IS_NOT_VALID);
+                return REDIRECT_ADD_MEMBER + "/" + teamId;
+            }
+        }
+
+        HttpEntity<String> request = createHttpEntityWithBody(addMemberForm.getEmails());
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity responseEntity;
+
+        try {
+            responseEntity = restTemplate.exchange(properties.addStudentsByEmail(teamId), HttpMethod.POST, request, String.class);
+        } catch (RestClientException e) {
+            log.warn("Error connecting to sio team service for adding members: {}", e);
+            redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+            return REDIRECT_ADD_MEMBER + "/" + teamId;
+        }
+
+        if (RestUtil.isError(responseEntity.getStatusCode())) {
+            String responseBody = responseEntity.getBody().toString();
+            String logPrefix = "Error in adding members to team " + teamId + ": {}";
+            MyErrorResource error;
+            String reason;
+            try {
+                error = objectMapper.readValue(responseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+                switch (exceptionState) {
+                    case TEAM_NOT_FOUND_EXCEPTION:
+                        reason = "Team " + teamId + " is not found.";
+                        log.warn(logPrefix, reason);
+                        redirectAttributes.addFlashAttribute(MESSAGE, reason);
+                        break;
+                    case DETERLAB_OPERATION_FAILED_EXCEPTION:
+                        log.warn(logPrefix, error.getMessage());
+                        redirectAttributes.addFlashAttribute(MESSAGE, error.getMessage());
+                        break;
+                    case USERNAME_ALREADY_EXISTS_EXCEPTION:
+                        reason = "User " + error.getMessage() + " already exists.";
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use already-defined constant 'USER_PREFIX' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)